arex-instrumentation-foundation/src/main/java/io/arex/foundation/serializer/gson/GsonRequestSerializer.java

```diff
@@ -0,0 +1,101 @@
+
+import com.google.auto.service.AutoService;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import io.arex.foundation.serializer.gson.adapter.DateAdapter;
+import io.arex.foundation.serializer.gson.adapter.GsonExclusion;
+import io.arex.foundation.serializer.gson.adapter.StringAdapter;
+import io.arex.foundation.serializer.gson.adapter.ProtobufAdapterFactory;
+import io.arex.foundation.serializer.gson.adapter.CalendarAdapter;
+import io.arex.foundation.serializer.gson.adapter.ClassAdapter;
+import io.arex.foundation.serializer.gson.adapter.DateTimeAdapter;
+import io.arex.foundation.serializer.gson.adapter.InstantAdapter;
+import io.arex.foundation.serializer.gson.adapter.JodaLocalDateAdapter;
+import io.arex.foundation.serializer.gson.adapter.JodaLocalDateTimeAdapter;
+import io.arex.foundation.serializer.gson.adapter.JodaLocalTimeAdapter;
+import io.arex.foundation.serializer.gson.adapter.LocalDateAdapter;
+import io.arex.foundation.serializer.gson.adapter.LocalDateTimeAdapter;
+import io.arex.foundation.serializer.gson.adapter.LocalTimeAdapter;
+import io.arex.foundation.serializer.gson.adapter.OffsetDateTimeAdapter;
+import io.arex.foundation.serializer.gson.adapter.TimeZoneAdapter;
+import io.arex.foundation.serializer.gson.adapter.XMLGregorianCalendarAdapter;
+import io.arex.inst.runtime.serializer.StringSerializable;
+import org.joda.time.DateTime;
+
+import javax.xml.datatype.XMLGregorianCalendar;
+import java.lang.reflect.Type;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.TimeZone;
+
+@AutoService(StringSerializable.class)
+public class GsonRequestSerializer implements StringSerializable {
+    private final Gson serializer;
+
+    public GsonRequestSerializer() {
+        this.serializer = new GsonBuilder()
+                .registerTypeAdapter(DateTime.class, new DateTimeAdapter.RequestSerializer())
+                .registerTypeAdapter(org.joda.time.LocalDateTime.class, new JodaLocalDateTimeAdapter.RequestSerializer())
+                .registerTypeAdapter(org.joda.time.LocalDate.class, new JodaLocalDateAdapter.Serializer())
+                .registerTypeAdapter(org.joda.time.LocalTime.class, new JodaLocalTimeAdapter.RequestSerializer())
+                .registerTypeAdapter(XMLGregorianCalendar.class, new XMLGregorianCalendarAdapter.RequestSerializer())
+                .registerTypeAdapter(Time.class, new DateAdapter.RequestSerializer())
+                .registerTypeAdapter(Timestamp.class, new DateAdapter.RequestSerializer())
+                .registerTypeAdapter(Date.class, new DateAdapter.RequestSerializer())
+                .registerTypeAdapter(java.sql.Date.class, new DateAdapter.RequestSerializer())
```

Comment:
can `new DateAdapter.RequestSerializer()`, `new CalendarAdapter.RequestSerializer()` be only one common instance?